diff --git a/sigviewer.pro b/sigviewer.pro
index d73ac51..2be0948 100644
--- a/sigviewer.pro
+++ b/sigviewer.pro
@@ -4,7 +4,7 @@ TARGET = sigviewer
 
 VERSION_MAJOR = 0
 VERSION_MINOR = 6
-VERSION_BUILD = 4
+VERSION_BUILD = 5
 
 DEFINES += \
     "VERSION_MAJOR=$$VERSION_MAJOR" \
@@ -33,7 +33,7 @@ CONFIG(release, debug|release) {
     UI_DIR = $$BUILD_DIR/release
 }
 
-CONFIG += warn_on link_prl qt thread c++11
+CONFIG += warn_on link_prl qt c++11
 
 macx {
     QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.9
diff --git a/src/gui_impl/signal_browser/adapt_browser_view_widget.ui b/src/gui_impl/signal_browser/adapt_browser_view_widget.ui
index 737d919..1229233 100644
--- a/src/gui_impl/signal_browser/adapt_browser_view_widget.ui
+++ b/src/gui_impl/signal_browser/adapt_browser_view_widget.ui
@@ -199,10 +199,10 @@
           <item>
            <widget class="QDoubleSpinBox" name="secsPerPageSpinbox">
             <property name="decimals">
-             <number>1</number>
+             <number>3</number>
             </property>
             <property name="minimum">
-             <double>0.100000000000000</double>
+             <double>0.00100000000000</double>
             </property>
            </widget>
           </item>
